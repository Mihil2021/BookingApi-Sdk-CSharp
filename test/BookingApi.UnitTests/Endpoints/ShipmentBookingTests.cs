using System.Threading.Tasks;
using BookingApi.Core.Api.Endpoints;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Shouldly;
using Xunit;

namespace BookingApi.UnitTests.Endpoints
{
    public class ShipmentBookingTests
    {
        [Fact]
        public async Task WhenCreatingRequest_ShouldMatchApproved()
        {
            // Arrange
            var request = new BookShipmentRequest("/")
                .WithReadyByDate()
                .WithHawb("123123123")
                .WithDescription("test Clothing")
                .WithDocuments()
                .WithPieces(pieces => pieces
                    .AddProduct(piece => piece
                        .Depth(1.00m)
                        .Height(1.00m)
                        .Width(1.00m)
                        .Weight(1.00m)
                        .NumberOfPieces(1)))
                .WithConsignee(consignee => consignee
                    .ContactName("Test")
                    .Company("Test")
                    .Address1("2 Willow Road")
                    .Zipcode("SL3 0BS")
                    .City("Slough")
                    .CountryCode("GB"))
                .WithServiceCode("IEL");

            // Act
            var bookingJson = JsonConvert.SerializeObject(request, Formatting.Indented, new JsonSerializerSettings
            {
                NullValueHandling = NullValueHandling.Ignore,
                Converters = new [] {
                    new StringEnumConverter()
                }
            });

            // Assert
            bookingJson.ShouldMatchApproved();
        }
    }
}
