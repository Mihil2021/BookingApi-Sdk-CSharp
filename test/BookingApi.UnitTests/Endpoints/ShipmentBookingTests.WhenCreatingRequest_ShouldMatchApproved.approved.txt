{
  "Pieces": [
    {
      "Depth": 1.00,
      "Height": 1.00,
      "Width": 1.00,
      "Weight": 1.00,
      "NumberOfPieces": 1
    }
  ],
  "ReadyByDate": "2021-01-25T14:48:50.9074036+00:00",
  "Hawb": "123123123",
  "Description": "test Clothing",
  "Value": 0.00,
  "NonDox": false,
  "DDP": false,
  "Pallet": false,
  "Consignee": {
    "ContactName": "Test",
    "Company": "Test",
    "Address1": "2 Willow Road",
    "City": "Slough",
    "Zipcode": "SL3 0BS",
    "CountryCode": "GB"
  },
  "Service": {
    "Code": "IEL"
  },
  "LabelFormat": "Pdf"
}
